src/screens/CurrencySelectScreen.tsx

```diff
@@ -0,0 +1,66 @@
+/*
+Copyright 2018 - 2022 The Alephium Authors
+This file is part of the alephium project.
+
+The library is free software: you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+The library is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with the library. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+import { StackScreenProps } from '@react-navigation/stack'
+import React from 'react'
+
+import BoxSurface from '~/components/layout/BoxSurface'
+import { ScreenSection } from '~/components/layout/Screen'
+import ScrollScreen from '~/components/layout/ScrollScreen'
+import RadioButtonRow from '~/components/RadioButtonRow'
+import { useAppDispatch, useAppSelector } from '~/hooks/redux'
+import RootStackParamList from '~/navigation/rootStackRoutes'
+import { currencySelected } from '~/store/settingsSlice'
+import { Currency } from '~/types/settings'
+import { currencies } from '~/utils/currencies'
+
+type ScreenProps = StackScreenProps<RootStackParamList, 'CurrencySelectScreen'>
+
+const currencyOptions = Object.values(currencies).map((currency) => ({
```

Comment:
Very nit: it seems like I'm a lodash propaganda master haha, but they have a handy `mapValues ` method. This native way is not so ugly though, just a bit longer.

Reply:
It doesn't return an array, however. For such a simple case I think I'll stick to vanilla JS 😉